dnsimple/dnsimple-support
# Add `search.js`

The support widget in `dnsimple-app` is consuming `search.json` from this repo and building out it's own results. This PR moves the article search/filter/sorting logic from the support widget to this repo. The point of doing this is to keep the logic in one place (and add a dictionary of search terms we should swap out for better results). In short, it exposes this (which can now be used by the support widget in `dnsimple-app`:

```js
window.DNSimpleSupport.search('dns')
```

Fixes https://github.com/dnsimple/dnsimple-support/issues/636

## 🔢 To The Reviewers

- I used old-browser-compatible JS syntax (eg. no es6).
- Remove `lunr` as a dependency.
- I moved search javascript to the default layout, allowing for live search from any page.
- Fixed a typo for a .md link of which `nanoc` was complaining.
- We did not have a JS testing thing in place, so I added `jest`.
- We did not have a JS linter thing in place, so I added `eslint`.
- I left `search.json` in tact so that the support widget in `dnsimple-app` keeps working until we update it to use the new search

## :mag: QA

- `yarn && yarn live`
- Visit http://localhost:3000/
- Search for "close", "delete domain", "twofa", etc – you should see reasonably good results along the way
- At some point, press <Enter> when you're in the search box – you should be sent to /search with the proper results
- Try to break it

## :shipit: Post tasks

- [ ] Ensure https://support.dnsimple.com/search.js doesn't do this: `â€™`. It seems to be a weird browser quirk when you visit the JS file directly – substitutes out `’`, but is fine through CURL and works correctly when loaded in HTML.

## :shipit: Verification

- [ ] https://support.dnsimple.com/ should have live searching
